### DISCUSSION TITLE
Picture In Picture window on button click

### DISCUSSION THREAD

**Discussion by kmurari103** [iTwin/itwinjs-core#2726] [Q&A] [answered]
Hi there,
I want to display a window/widget pane when toolbar button is clicked (like a window on top of the viewport), this window would have a dropdown and attached iframe which would show some graph/data based on selection in dropdown. I am using **"@itwin/web-viewer-react": "^1.3.1",**. Is there any samples/snippets on how to achieve this ? Any help is appreciated.
Here's a raw mockup of what i mean with PnP window, i would click toolbar button just right to home button & it should popup a new window on top of viewport with above intended functionality. 
![image](https://user-images.githubusercontent.com/27068008/142144548-087c630d-a2ae-47b2-aa85-7878ad0ef5d0.png)

**Comment by williamkbentley** [comment, marked as answer]
You're going to need a `UiItemsProvider` and implement `provideToolbarButtonItems`. Then you can provide any react control to the button as a `popupPanelNode`. For example: https://www.itwinjs.org/sandboxes/williamkbentley/content-manip-select-iframe

**Comment by kmurari103** [comment]
Thanks @williamkbentley for the help, just a follow up query- is it possible to make this popup panel resizable? I mean dragging in x-direction should increase/decrease its width & y-direction should increase/decrease its height.

**Reply by bsteinbk** [reply]
@kmkashyap The popup panel is not currently resizable. @NancyMcCallB this may be something to add to our backlog.

**Reply by williamkbentley** [reply]
Another caveat to keep in mind is the state of the popup panel is reset every time it is closed. So you will lose the state of the <Select> in that component (unless you save it off in Redux or some other state manager). If these two things are limiting for your use case, maybe the content manipulation tools and  `popupPanelNode` is not the right approach.

**Reply by bsteinbk** [reply]
The button could also open a Modeless or Modal dialog which does support resizing. In that case too, you would have to save and retrieve the size from redux or local storage yourself.
